CMakeLists.txt

```diff
@@ -492,10 +495,27 @@ foreach (test IN ITEMS path str)
 endforeach ()
 
 if ("cxx_relaxed_constexpr" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
-   add_executable(mlib.int128.test src/mlib/int128.test.cpp src/mlib/int128.test.c)
-   add_test (mlib.int128 mlib.int128.test)
-   target_link_libraries (mlib.int128.test PRIVATE mongo::mlib Threads::Threads)
-   target_compile_features (mlib.int128.test PRIVATE cxx_relaxed_constexpr)
+   file (GLOB_RECURSE test_files CONFIGURE_DEPENDS src/*.test.cpp)
```

Comment:
The C driver build on clang 3.4 on Ubuntu 14.04 fails to build (see [patch](https://spruce.mongodb.com/task/mongo_c_driver_clang34ubuntu_debug_compile_sasl_openssl_cse_patch_a8560d7488778e480b7c61a0f8931ddbcc310c7d_63c6c96fd6d80a4e2436bcd1_23_01_17_16_14_40/logs?execution=0)):

```
[2023/01/17 16:18:22.741] In file included from /data/mci/10936fccdaa1b5078d457ecaae4d8c66/mongoc/libmongocrypt/src/mlib/int128.test.cpp:3:
[2023/01/17 16:18:22.741] In file included from /data/mci/10936fccdaa1b5078d457ecaae4d8c66/mongoc/libmongocrypt/src/mlib/./check.hpp:4:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/iostream:39:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ostream:38:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ios:42:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/ios_base.h:41:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/locale_classes.h:40:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/string:52:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/basic_string.h:2815:
[2023/01/17 16:18:22.741] In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ext/string_conversions.h:43:
[2023/01/17 16:18:22.741] /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/cstdio:120:11: error: no member named 'gets' in the global namespace
[2023/01/17 16:18:22.741]   using ::gets;
[2023/01/17 16:18:22.741]         ~~^
```

Since libmongocrypt is not tested on Ubuntu 14.04, I think I will remove the `-cse` task on Ubuntu 14.04. This may not be worth investigating. Commenting in case this points to an issue in the required C++ compiler conditions.

Reply:
This compiler error is a real blast from the past...

Reply:
The C driver can work around this with the addition of https://github.com/mongodb/libmongocrypt/pull/532.